src/Twig/UnitsExtension.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Twig;
+
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class UnitsExtension extends AbstractExtension
+{
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('is_imperial', [$this, 'isImperial']),
+            new TwigFunction('m_or_f', [$this, 'm_or_f']),
+            new TwigFunction('kph_or_mph', [$this, 'kph_or_mph']),
+            new TwigFunction('km_or_mil', [$this, 'km_or_mil']),
+        ];
+    }
+
+    public function isImperial(string $locale): bool
+    {
+        // TODO cookies
+        // TODO User
+
+        return 'en' === $locale;
+    }
+
+    public function m_or_f(bool $isImperial, int $value): string
+    {
+        if ($isImperial) {
+            return round($value * 3.281).' ft';
+        }
+
+        return $value.' m';
+    }
+
+    public function kph_or_mph(bool $isImperial, int $value): string
+    {
+        if ($isImperial) {
+            return round($value / 1.609).' mph';
+        }
+
+        return $value.' km/h';
+    }
+
+    public function km_or_mil(bool $isImperial, int $value): string
+    {
+        if ($isImperial) {
+            return round($value / 1.609).' mil';
```

Comment:
official abbreviation for mile seems to be "mi"

https://en.wikipedia.org/wiki/Mile#:~:text=than%20Roman%20miles.-,Abbreviation,)%20and%20millilitre%20(ml).

Reply:
Changed!

Reply:
The main problem was the translation I think. I need to display a different sentence depending on the language ("8 mi away" in english, "à 8 km" in french for example), so I updated the convert functions.

They now return an array with the value and the units, the two types of data can be used differently.